### PR TITLE
Add cluster config for e2e setup with Libra node

### LINKED ISSUE #858: [e2e] Skip all shard check
when connecting with Libra, only one libra chain sends out minor block, therefore we have to skip `master.__has_all_shards()` which is used for master has successfully connected with all shards.

### PR DESCRIPTION
**Pull request by FreshmanQ**
resolves #858, #859, #860.

### CODE REVIEW

File: quarkchain/cluster/master.py [hunk 1]
```diff
@@ -1064,9 +1064,10 @@ def __log_summary(self):
     async def __init_cluster(self):
         await self.__connect_to_slaves()
         self.__log_summary()
-        if not self.__has_all_shards():
-            Logger.error("Missing some shards. Check cluster config file!")
-            return
+        if self.slave_pool:
```
Comment (ninjaahhh) [inline quarkchain/cluster/master.py]: line 1065 to line 1071 are all specific to qkcrpc. your condition check can be added on top of all of them

File: quarkchain/cluster/root_state.py [hunk 2]
```diff
@@ -456,6 +456,10 @@ def validate_block(self, block, adjusted_diff: int = None):
                 block.header.height,
             )
             actual_coinbase_amount = block.header.coinbase_amount_map.balance_map
+
+            if self.env.cluster_config.GRPC_SLAVE_LIST:
```
Comment (ninjaahhh) [inline quarkchain/cluster/root_state.py]: for this check and the one at line 493, I suggest adding two flags at the method parameters `skip_coinbase_check` and `skip_root_block_linkage`

File: quarkchain/cluster/root_state.py [hunk 3]
```diff
@@ -517,12 +525,12 @@ def validate_block(self, block, adjusted_diff: int = None):
             check(len(headers) > 0)
 
             shard_config = self.env.quark_chain_config.shards[full_shard_id]
-            if len(headers) > shard_config.max_blocks_per_shard_in_one_root_block:
-                raise ValueError(
-                    "too many minor blocks in the root block for shard {}".format(
-                        full_shard_id
-                    )
-                )
+            # if len(headers) > shard_config.max_blocks_per_shard_in_one_root_block:
```
Comment (ninjaahhh) [inline quarkchain/cluster/root_state.py]: this should still be enforced?

File: quarkchain/cluster/root_state.py [hunk 4]
```diff
@@ -542,7 +550,10 @@ def validate_block(self, block, adjusted_diff: int = None):
             else:
                 headers = [prev_header_in_last_root_block] + headers
             for i in range(len(headers) - 1):
+                temp1 = headers[i + 1].hash_prev_minor_block
```
Comment (ninjaahhh) [inline quarkchain/cluster/root_state.py]: remove debug logging

File: quarkchain/cluster/root_state.py (hunk 2 above)
Comment (FreshmanQ) [reply quarkchain/cluster/root_state.py]: ok, then libra should not include prev_root_block_id in Minorblockheader

File: quarkchain/cluster/root_state.py [hunk 5]
```diff
@@ -429,7 +429,9 @@ def is_same_chain(self, longer_block_header, shorter_block_header):
             header = self.db.get_root_block_header_by_hash(header.hash_prev_block)
         return header == shorter_block_header
 
-    def validate_block(self, block, adjusted_diff: int = None):
+    def validate_block(
+        self, block, adjusted_diff: int = None, skip_root_block_linkage: bool = False
```
Comment (ninjaahhh) [inline quarkchain/cluster/root_state.py]: nit: `skip_check_root_block_linkage`

**Comment by FreshmanQ** [comment]
> has this PR updated max number of minor block headers in a root block?

yes, it's already been set in cluster_config_libra.json

File: devnet/singularity/cluster_config_libra.json [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+        "EXTRA_SHARD_BLOCKS_IN_ROOT_BLOCK": 97,
+        "POSW_CONFIG": {
+          "ENABLED": false,
+          "DIFF_DIVIDER": 20,
+          "WINDOW_SIZE": 256,
+          "TOTAL_STAKE_PER_BLOCK": 0
+        },
+        "EPOCH_INTERVAL": 3153600
+      },
+      {
```
Comment (ninjaahhh) [inline devnet/singularity/cluster_config_libra.json]: we don't need chain 1 through chain 7, right?

File: devnet/singularity/cluster_config_libra.json [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+          "VERSION": 0,
+          "HEIGHT": 0,
+          "HASH_PREV_MINOR_BLOCK": "0000000000000000000000000000000000000000000000000000000000000000",
+          "HASH_MERKLE_ROOT": "0000000000000000000000000000000000000000000000000000000000000000",
+          "EXTRA_DATA": "497420776173207468652062657374206f662074696d65732c206974207761732074686520776f727374206f662074696d65732c202e2e2e202d20436861726c6573204469636b656e73",
+          "TIMESTAMP": 1556639999,
+          "DIFFICULTY": 5000000000,
+          "GAS_LIMIT": 12000000,
+          "NONCE": 0,
+          "ALLOC": {
```
Comment (ninjaahhh) [inline devnet/singularity/cluster_config_libra.json]: we probably also don't need `ALLOC` here

File: quarkchain/cluster/master.py [hunk 8]
```diff
@@ -1063,11 +1063,12 @@ def __log_summary(self):
 
     async def __init_cluster(self):
         await self.__connect_to_slaves()
-        self.__log_summary()
-        if not self.__has_all_shards():
-            Logger.error("Missing some shards. Check cluster config file!")
-            return
-        await self.__setup_slave_to_slave_connections()
+        if not self.grpc_slave_pool:
```
Comment (ninjaahhh) [inline quarkchain/cluster/master.py]: 1. should check on `self.slave_pool`
2. `self.__connect_to_slaves()` and `self.__init_shards()` should also be under the condition check?

Comment (FreshmanQ) [reply quarkchain/cluster/master.py]: `self.slave_pool` is generated by `self.__connect_to_slaves()`

Comment (ninjaahhh) [reply quarkchain/cluster/master.py]: then how about check `self.cluster_config.get_slave_info_list()`

**Pull request merged**